### PR TITLE
Neutralizer/adding tests

### PR DESCRIPTION
**Pull request by Neutralizer**
Tests Ready

### CODE REVIEW

File: src/test/java/com/books/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.Optional;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("dev")
```
Comment (matrixuser87) [inline src/test/java/com/books/BookControllerTest.java]: remove the profile. you want those test to run in all environments

File: src/test/java/com/books/BookRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.springframework.test.context.ContextConfiguration;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ActiveProfiles("dev")
```
Comment (matrixuser87) [inline src/test/java/com/books/BookRepositoryTest.java]: remove this line

File: src/test/java/com/books/BookRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        Book book = found.get(0);
+        book.setTitle("Updated Title");
+
+        testEntityManager.persist(book);
+        testEntityManager.flush();
+
+        Optional<Book> updated = bookRepository.findById(book.getId());
+
+        assertThat(updated.orElseThrow(NoSuchElementException::new).getTitle(),
```
Comment (matrixuser87) [inline src/test/java/com/books/BookRepositoryTest.java:89]: java 8.. nice :D

File: src/test/java/com/books/BookRepositoryTest.java (hunk 2 above)
Comment (Neutralizer) [reply src/test/java/com/books/BookRepositoryTest.java]: Done

File: src/test/java/com/books/BookControllerTest.java (hunk 1 above)
Comment (Neutralizer) [reply src/test/java/com/books/BookControllerTest.java]: Done

**Pull request merged**